### PR TITLE
GEODE-8881: add tests for redis HKEYS command

### PR DESCRIPTION
**Pull request by nonbinaryprogrammer**
- add unit/integration tests
- add DUnit concurrency tests
- update the docs

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/hash/HKeysDUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+    jedis1.hset(key, testMap);
+
+    new ConcurrentLoopingThreads(NUM_ITERATIONS,
+        (i) -> {
+          int newIndex = HASH_SIZE + i;
+          jedis1.hset(key, "field-" + newIndex, "value-" + newIndex);
+          expectedFields.add("field-" + newIndex);
+        },
+        (i) -> assertThat(jedis2.hkeys(key)).containsExactlyInAnyOrderElementsOf(expectedFields));
```
Comment (jdeppe-pivotal) [inline geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/hash/HKeysDUnitTest.java:105]: This and the other uses of `ConcurrentLoopingThreads` are missing the `.run()` call to actually execute them. Unfortunately you'll see that they cause a `ConcurrentModificationException` on the `expectedFields` variable. Even so, there would be a race between lines 102 and 103 and the assertion on 105 that does the check on the hash. It's always hard to know what to assert on in this kind of a case. You could probably just adjust the assertion to be `containsAll()`.

**Review by jdeppe-pivotal — CHANGES_REQUESTED** [review]

**Review by jdeppe-pivotal — APPROVED** [review]

File: geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/hash/HKeysDUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    Map<String, String> testMap = makeHashMap(NUM_ITERATIONS, "field-", "value-");
+
+    jedis1.hset(key, testMap);
+
+    new ConcurrentLoopingThreads(NUM_ITERATIONS,
+        (i) -> jedis1.hset(key, "field-" + i, "changedValue-" + i),
+        (i) -> assertThat(jedis2.hkeys(key)).containsExactlyInAnyOrderElementsOf(testMap.keySet()))
+            .run();
+
+    assertThat(jedis1.hkeys(key)).containsExactlyInAnyOrderElementsOf(testMap.keySet());
```
Comment (ringles) [inline geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/hash/HKeysDUnitTest.java:146]: Technically, this test doesn't confirm that the values were actually changed by the thread on line 142. It just confirms the keys are all still there. A loop or assert checking that the new values are all "changedValue-[number]" might be useful.

**Review by ringles — CHANGES_REQUESTED** [review]
One suggestion, otherwise good.

File: geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/hash/HKeysDUnitTest.java (hunk 2 above)
Comment (nonbinaryprogrammer) [reply geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/hash/HKeysDUnitTest.java:149]: 👍

**Pull request merged**